src/web/WebSession.C

```diff
@@ -680,9 +680,16 @@ bool WebSession::start(WebResponse *response)
 {
   try {
     app_ = controller_->doCreateApplication(this).release();
-    if (!app_->internalPathValid_)
-      if (response->responseType() == WebResponse::ResponseType::Page)
-        response->setStatus(404);
+    if (app_) {
+      if (!app_->internalPathValid_) {
+        if (response->responseType() == WebResponse::ResponseType::Page) {
+          response->setStatus(404);
+        }
+      }
+    }
```

Comment:
Small style comment: in the Wt source code we put the `else` on the same line as the closing bracket, so like this:

```cpp
} else {
```

Reply:
Hi Roel,

you're welcome. I've just updated the commit according to your remarks.

Steven